x/ccv/provider/keeper/msg_server.go

```diff
@@ -126,3 +129,72 @@ func (k msgServer) RegisterConsumerRewardDenom(goCtx context.Context, msg *types
 
 	return &types.MsgRegisterConsumerRewardDenomResponse{}, nil
 }
+
+func (k msgServer) SubmitConsumerMisbehaviour(goCtx context.Context, msg *types.MsgSubmitConsumerMisbehaviour) (*types.MsgSubmitConsumerMisbehaviourResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := k.Keeper.HandleConsumerMisbehaviour(ctx, *msg.Misbehaviour); err != nil {
+		return &types.MsgSubmitConsumerMisbehaviourResponse{}, err
```

Comment:
nit: `return nil, err` to be consistent with what we do in ` SubmitConsumerDoubleVoting`